python/vast/vast/cli.py

```diff
@@ -65,17 +66,21 @@ def run(*args, stdin) -> subprocess.Popen:
 
     def __getattr__(self, name, **kwargs):
         """Chains every unknown method call to the internal call stack."""
+        # Don't break `__special__` attributes.
+        if name.startswith("__"):
+            raise AttributeError(name)
+        new = copy.deepcopy(self)
         if name.endswith("_"):
             # trim trailing underscores to overcome the 'import' keyword
             name = name[:-1]
         if not name == "__iter_":
-            self.args.append(name.replace("_", "-"))
+            new.args.append(name.replace("_", "-"))
 
         def command(*args, **kwargs):
             if kwargs:
-                self.args.extend(CLI.arguments(**kwargs))
+                new.args.extend(CLI.arguments(**kwargs))
             if args:
-                self.args.extend(args)
-            return self
+                new.args.extend(args)
+            return new
```

Comment:
what problem does this solve exactly?

Reply:
Every call into this function creates a new copy of `Cli` instead of mutating the object in place. Without this a `Cli` object is not reusable as a second invocation would just append commands an options to the command line that was set in the previous call.
Example:
```
>>> mycli = Cli(endpoint="127.0.0.1:42000")
>>> mycli.import().zeek().exec()
vast -e 127.0.0.1:42000 import zeek
>>> mycli.status().exec()
vast -e 127.0.0.1:42000 import zeek status
```

Reply:
I asked myself the same question looking at this. It would be good to document.